src/Dictionaries/HashedArrayDictionary.h

```diff
@@ -185,6 +195,22 @@ class HashedArrayDictionary final : public IDictionary
 
     void calculateBytesAllocated();
 
+    UInt64 getShard(UInt64 key) const
+    {
+        if constexpr (!sharded)
+            return 0;
+        /// NOTE: function here should not match with the DefaultHash<> since
+        /// it used for the HashMap/sparse_hash_map.
+        return intHashCRC32(key) % configuration.shards;
+    }
+
+    UInt64 getShard(StringRef key) const
+    {
+        if constexpr (!sharded)
+            return 0;
+        return StringRefHash()(key) % configuration.shards;
```

Comment:
I used the same logic as in `HashedDictionary.h`, but indeed it can be an issue (even comment above mentions it). I'll try to check if it should and can be improved.